### PR TITLE
Build tasks: Move review app tasks into `app/tasks`

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR splits out all "Review app" tasks into `app/tasks` as part of:

* https://github.com/alphagov/govuk-frontend/issues/2886

This reduces the risk of affecting `govuk-frontend` output by keeping each build separate:

1. **Project-level** `./tasks` via [gulpfile.mjs](https://github.com/alphagov/govuk-frontend/blob/build-tasks-gulpfile-app/gulpfile.mjs)
2. **Review app** `./app/tasks` via [app/gulpfile.mjs](https://github.com/alphagov/govuk-frontend/blob/build-tasks-gulpfile-app/app/gulpfile.mjs)

```console
├── app
│   ├── dist
│   ├── src
│   └── tasks
└── tasks
```

## Creating tasks

Whilst Gulp is still used as a task runner, I've switched to the [export your tasks](https://gulpjs.com/docs/en/getting-started/creating-tasks) pattern vs [using `gulp.task()`](https://gulpjs.com/docs/en/api/task/)

>**Reminder:** This API isn't the recommended pattern anymore - [export your tasks](https://gulpjs.com/docs/en/getting-started/creating-tasks).

All our new tasks are now plain async functions:

```mjs
await files.copy('**/*', {
  srcPath: join(paths.src, 'govuk/assets'),
  destPath: join(paths.app, 'dist/assets')
})
```

But they can be composed into tasks for Gulp's `gulp.series()` and `gulp.parallel()` etc

```mjs
task.name('copy:assets', () => files.copy('**/*', {
  srcPath: join(paths.src, 'govuk/assets'),
  destPath: join(paths.app, 'dist/assets')
}))
```

### CODE REVIEW

**Comment by romaricpascal** [comment]
@colinrotherham I’m getting lost between what’s for moving the tasks to the app proper vs. what’s for moving Gulp tasks to using functions (if I understood correctly what’s going on), sorry. Would it be possible to split these two aspects to ease the review, please (let me know if it’s too much of a pain )?

**Comment by colinrotherham** [comment]
> @colinrotherham I’m getting lost between what’s for moving the tasks to the app proper vs. what’s for moving Gulp tasks to using functions (if I understood correctly what’s going on), sorry. Would it be possible to split these two aspects to ease the review, please (let me know if it’s too much of a pain )?

It was a massive pain but it's far better now 😆

Let me know if you'd like things splitting out more. Themes included:

1. Updating npm task runner for workspaces
2. Updating task and script documentation
3. Removing `gulp.task()` wrapper scripts where possible
4. Replacing Heroku script with `npm start --workspace app`
5. Bug fix for Nodemon to ignore frontend `*.mjs` changes

Anything that was discovered during the move

**Review by romaricpascal — APPROVED** [review]
I think it's almost there, just had a couple of questions and little things in the comments 😄 Cheers for the move to plain functions with `task.name` and the genral tidy up.

File: jest-dev-server.config.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ const { ports } = require('./config/index.js')
  * @type {import('jest-dev-server').Config}
  */
 module.exports = {
-  command: 'npm start --workspace app',
+  // Start Express.js without "prestart" build script
+  command: 'npm start --ignore-scripts --workspace app',
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply jest-dev-server.config.js:8]: This is a learning from the **GOV.UK Design System**
https://github.com/alphagov/govuk-design-system/blob/main/jest-puppeteer.config.js#L6

Every time you run `npm test` or `npx jest ./path/to/spec.test.js` it does a full rebuild each time

So here we cheat and give Jest permission to run what's already built

For anyone else running `npm start` I'd like it to stay nice and simple 😊 

Well spotted though

File: app/package.json [hunk 2]
```diff
@@ -9,7 +9,10 @@
   },
   "license": "MIT",
   "scripts": {
+    "build": "gulp build",
+    "dev": "gulp dev",
     "serve": "nodemon",
+    "prestart": "npm run build",
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply app/package.json:15]: Would you mind if we kept `npm start`?

Keeps things simple

Although perhaps in future should align `start`/`dev`?

### Server start

```console
npm start
npm start --workspace app
npm start --workspace docs/examples/webpack
```

### Development

```console
npm run dev
npm run dev --workspace app
npm run dev --workspace docs/examples/webpack
```

But for now we have a necessary mismatch with `npm start` at project-level:

```console
npm run dev --workspace app
```

File: .github/workflows/screenshots.yml [hunk 3]
```diff
@@ -41,7 +41,7 @@ jobs:
         uses: ./.github/workflows/actions/build
 
       - name: Start review application
-        run: npm run serve &
+        run: npm start --workspace app &
```
Comment (colinrotherham) [inline .github/workflows/screenshots.yml:44]: Need to make sure we skip the `prestart` build here (as it's cached in GitHub Actions)

```suggestion

        # Start Express.js without "prestart" build script
        run: npm start --ignore-scripts --workspace app &
```

**Pull request merged**